internal/v1/staffing/main.go

```diff
@@ -0,0 +1,53 @@
+package staffing
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/adh-partnership/api/pkg/database/dto"
+	"github.com/adh-partnership/api/pkg/database/models"
+	"github.com/adh-partnership/api/pkg/discord"
+	"github.com/adh-partnership/api/pkg/gin/middleware/auth"
+	"github.com/adh-partnership/api/pkg/gin/response"
+	"github.com/gin-gonic/gin"
+	"istio.io/pkg/log"
+)
+
+func Routes(r *gin.RouterGroup) {
+	r.POST("", auth.NotGuest, requestStaffing)
+}
+
+// Submit a staffing request.
+// @Summary Submit a staffing request
+// @Description Submit a staffing request
+// @Tags Staffing
+// @Param data body dto.StaffingRequest true "Request Data"
+// @Success 202
+// @Failure 400 {object} response.R "Invalid form submission"
+// @Failure 401 {object} response.R "Not logged in"
+// @Failure 500 {object} response.R
+// @Router /v1/staffing/ [post]
+func requestStaffing(c *gin.Context) {
+	user := c.MustGet("x-user").(*models.User)
+
+	var dto dto.StaffingRequest
+	if err := c.ShouldBindJSON(&dto); err != nil {
```

Comment:
Since the API supports more than JSON, you should be using `c.ShouldBind` here and allow gin to determine the body format.